_aggregations/metric/geocentroid.md

```diff
@@ -109,23 +105,111 @@ GET /restaurants/_search
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 4675,
+      "relation": "eq"
+    },
+    "max_score": null,
+    "hits": []
+  },
+  "aggregations": {
+    "continents": {
+      "doc_count_error_upper_bound": 0,
+      "sum_other_doc_count": 0,
+      "buckets": [
+        {
+          "key": "Asia",
+          "doc_count": 1220,
+          "centroid": {
+            "location": {
+              "lat": 28.023606536509163,
+              "lon": 47.83377046025068
+            },
+            "count": 1220
+          }
+        },
+        {
+          "key": "North America",
+          "doc_count": 1206,
+          "centroid": {
+            "location": {
+              "lat": 39.06542286878007,
+              "lon": -85.36152573149485
+            },
+            "count": 1206
+          }
+        },
+        {
+          "key": "Europe",
+          "doc_count": 1172,
+          "centroid": {
+            "location": {
+              "lat": 48.125767892293325,
+              "lon": 2.7529009746915243
+            },
+            "count": 1172
+          }
+        },
+        {
+          "key": "Africa",
+          "doc_count": 899,
+          "centroid": {
+            "location": {
+              "lat": 30.780756367941297,
+              "lon": 13.464182392125318
+            },
+            "count": 899
+          }
+        },
+        {
+          "key": "South America",
+          "doc_count": 178,
+          "centroid": {
+            "location": {
+              "lat": 4.599999985657632,
+              "lon": -74.10000007599592
+            },
+            "count": 178
+          }
+        }
+      ]
+    }
+  }
+}
+```
+</details>
 
-When a document contains multiple geopoint values in a field, the `geohash_grid` aggregation assigns the document to multiple buckets, even if one or more of its geopoints are outside the bucket boundaries. This behavior is different from how individual geopoints are treated, where only those within the bucket boundaries are considered.
+### Example: Nesting under a `geohash_grid` aggregation
 
-When you nest the `geo_centroid` aggregation under the `geohash_grid` aggregation, each centroid is calculated using all geopoints in a bucket, including those that may be outside the bucket boundaries. This can result in centroid locations that fall outside the geographic area represented by the bucket.
+The [`geohash_grid`]({{site.url}}{{site.baseurl}}/aggregations/bucket/geohash-grid/)
+aggregation partitions geospatial data into buckets based on [geohash](https://en.wikipedia.org/wiki/Geohash) prefixes.
 
-#### Example 
+If a document contains multiple geopoint values in a field, the `geohash_grid` aggregation can assign the entire document to multiple buckets, even if one or more of its geopoints are outside the bucket boundaries. In other words, an entire document is assigned into a bucket if one or more of its points falls in that bucket.
```

Comment:
My assumption is that each document is put in a given bucket only once, even if it has multiple points in that bucket. Please confirm.

Reply:
geohash_grid doesn't belong here I suppose. 
From below example request and response, seems one document can contain multiple points, then one document could fall under more than one bucket, because the points fall under different buckets.

Reply:
@bowenlan-amzn Exactly. But, each document can fall into a given bucket only once, is that right?

Reply:
yeah, if look at any certain bucket, it should only have any document once for sure.
How it works in code at runtime is every document is processed once, and it process every point in this document in a loop, to decide which bucket the point belongs.